Formula/opentsdb.rb

```diff
@@ -0,0 +1,151 @@
+class HbaseLZORequirement < Requirement
+  fatal true
+
+  satisfy(:build_env => false) { !Tab.for_name("hbase").without?("lzo") }
+
+  def message; <<-EOS.undent
+  hbase must not have disabled lzo compression to use it in opentsdb:
+    brew install hbase
+    not
+    brew install hbase --without-lzo
+  EOS
+  end
+end
+
+class Opentsdb < Formula
+  desc "Scalable, distributed Time Series Database."
+  homepage "http://opentsdb.net/"
+  url "https://github.com/OpenTSDB/opentsdb/releases/download/v2.2.0/opentsdb-2.2.0.tar.gz"
+  sha256 "5689d4d83ee21f1ce5892d064d6738bfa9fdef99f106f45d5c38eefb9476dfb5"
+
+  option "without-lzo", "Don't use LZO Compression"
+
+  if build.without? "lzo"
+    depends_on "hbase"
+  else
+    depends_on "hbase"
+    depends_on "lzo"
+    depends_on HbaseLZORequirement
+  end
+  depends_on :java => "1.6+"
+  depends_on "gnuplot" => :optional
+
+  def install
+    # submitted to upstream https://github.com/OpenTSDB/opentsdb/pull/711
+    # mkdir_p is called from in a subdir of build so needs an extra ../ and there is no rule to create $(classes) and
+    # everything builds without specifying them as dependencies of the jar.
+    inreplace "Makefile.in" do |s|
+      s.sub!(/(\$\(jar\): manifest \.javac-stamp) \$\(classes\)/, '\1')
+      s.sub!(/(echo " \$\(mkdir_p\) '\$\$dstdir'"; )/, '\1../')
+    end
+
+    mkdir "build" do
+      system "../configure",
+             "--disable-silent-rules",
+             "--prefix=#{prefix}",
+             "--mandir=#{man}",
+             "--sysconfdir=#{etc}",
+             "--localstatedir=#{var}/opentsdb"
+      system "make"
+      bin.mkpath
+      system "make", "install-exec-am", "install-data-am"
+    end
+
+    env = Language::Java.java_home_env.merge(:HBASE_HOME => Formula["hbase"].opt_libexec, :COMPRESSION => (build.with?("lzo") ? "LZO" : "NONE"))
+    (bin/"create_table.sh").write_env_script opt_pkgshare/"tools/create_table.sh", env
+
+    inreplace pkgshare/"etc/opentsdb/opentsdb.conf", "/usr/share", "#{HOMEBREW_PREFIX}/share"
+    etc.install Dir["#{pkgshare}/etc/opentsdb"]
+    (pkgshare/"plugins").mkpath
+    (pkgshare/"plugins/.keep").write ""
+  end
+
+  def post_install
+    (var/"cache/opentsdb").mkpath
+
+    system "#{Formula["hbase"].opt_libexec}/bin/start-hbase.sh"
+    system "#{bin}/create_table.sh"
+    system "#{Formula["hbase"].opt_libexec}/bin/stop-hbase.sh"
+  end
+
+  plist_options :manual => "tsdb tsd --config=#{HOMEBREW_PREFIX}/etc/opentsdb/opentsdb.conf "\
+                           "--staticroot=#{HOMEBREW_PREFIX}/opt/opentsdb/share/opentsdb/static/ "\
+                           "--cachedir=#{HOMEBREW_PREFIX}/var/cache/opentsdb --port=4242 "\
+                           "--zkquorum=localhost:2181 --zkbasedir=/hbase --auto-metric"
```

Comment:
Should this `tsdb` be `#{opt_bin}/tsdb` instead? Given the long list of arguments here I wonder if we can refactor this to avoid duplicating all those arguments here, in the `plist` method, and in the `test do` block. I'm thinking something along the lines of:

```ruby
def tsdb_args
  %W[
    #{opt_bin}/tsdb
    tsd
    --config=#{HOMEBREW_PREFIX}/etc/opentsdb/opentsdb.conf
    --staticroot=#{HOMEBREW_PREFIX}/opt/opentsdb/share/opentsdb/static/
    --cachedir=#{HOMEBREW_PREFIX}/var/cache/opentsdb
    --port=4242
    --zkquorum=localhost:2181
    --zkbasedir=/hbase
    --auto-metric
  ]
end
```

It could be used here as follows:

```ruby
plist_options :manual => tsdb_args.join(" ")
```

And in the `plist` method:

```ruby
def plist
  program_args = tsdb_args.map { |s| "    <string>#{s}</string>" }.join("\n")
  <<-EOS.undent
    … snip …
      <key>ProgramArguments</key>
      <array>
    #{program_args}
      </array>
    … snip …
  EOS
end
```